services/web/server/src/simcore_service_webserver/projects/projects_handlers.py

```diff
@@ -53,104 +54,135 @@
                 )
+        if source_project:
             # clone template as user project
-            project, nodes_map = clone_project_document(
-                source_project, forced_copy_project_id=None
+            new_project, nodes_map = clone_project_document(
+                source_project,
+                forced_copy_project_id=None,
+                clean_output_data=(copy_data == False),
             )
-            clone_data_coro = copy_data_folders_from_project(
-                request.app, source_project, project, nodes_map, user_id
+            if template_uuid:
+                # remove template access rights
+                new_project["accessRights"] = {}
+            # the project is to be hidden until the data is copied
+            hidden = copy_data
+            clone_data_coro = (
+                copy_data_folders_from_project(
+                    request.app, source_project, new_project, nodes_map, user_id
+                )
+                if copy_data
+                else None
             )
-
-            # remove template access rights
-            project["accessRights"] = {}
             # FIXME: parameterized inputs should get defaults provided by service
 
         # overrides with body
         if request.has_body:
             predefined = await request.json()
-            if project:
+            if new_project:
                 for key in OVERRIDABLE_DOCUMENT_KEYS:
                     non_null_value = predefined.get(key)
                     if non_null_value:
-                        project[key] = non_null_value
+                        new_project[key] = non_null_value
             else:
                 # TODO: take skeleton and fill instead
-                project = predefined
+                new_project = predefined
 
         # re-validate data
-        projects_api.validate_project(request.app, project)
+        await projects_api.validate_project(request.app, new_project)
 
         # update metadata (uuid, timestamps, ownership) and save
-        project = await db.add_project(
-            project,
+        new_project = await db.add_project(
+            new_project,
             user_id,
             force_as_template=as_template is not None,
             hidden=hidden,
         )
 
         # copies the project's DATA IF cloned
         if clone_data_coro:
-            await clone_data_coro
+            assert source_project  # nosec
+            if as_template:
+                # we need to lock the original study while copying the data
+                async with projects_api.lock_with_notification(
+                    request.app,
+                    source_project["uuid"],
+                    ProjectStatus.CLONING,
+                    user_id,
+                    await get_user_name(request.app, user_id),
+                ):
+
+                    await clone_data_coro
+            else:
+                await clone_data_coro
```

Comment:
QUESTION? why dont' you lock with notification when ``not as_template``? I guess this is creating a normal project so the user in principle does not see in the UI the thumbnail of the dashboard. Nonetheless, logically it should also lock? what would happen in one day the GUI changes? that will lead to an inconsistent state?

Reply:
so the lock is here to prevent people from modifying a project while it is accessed.
in the case of ```from_template``` we copy from a template (which is a read-only snapshot per se) to a new study. Per construction the template cannot be modified, therefore there is no need to lock it.
Now that I think of it, we could actually lock the new study until it is complete... maybe something to think about.